kong.conf.default

```diff
@@ -415,26 +415,42 @@
 # they have been set.
 
 #dns_resolver =                  # Comma separated list of nameservers, each
-                                 # entry in `ipv4[:port]` format to be used by
+                                 # entry in `ip[:port]` format to be used by
                                  # Kong. If not specified the nameservers in
                                  # the local `resolv.conf` file will be used.
-                                 # Port defaults to 53 if omitted.
+                                 # Port defaults to 53 if omitted. Accepts
+                                 # both IPv4 and IPv6 addresses.
 
 #dns_hostsfile = /etc/hosts      # The hosts file to use. This file is read
                                  # once and its content is static in memory.
                                  # To read the file again after modifying it,
                                  # Kong must be reloaded.
 
-#dns_order = LAST,SRV,A,CNAME    # the order in which to resolve different
+#dns_order = LAST,SRV,A,CNAME    # The order in which to resolve different
                                  # record types. The `LAST` type means the
                                  # type of the last successful lookup (for the
                                  # specified name). The format is a (case
                                  # insensitive) comma separated list.
 
-#dns_not_found_ttl = 30.0        # ttl in seconds for empty DNS responses and
+#dns_stale_ttl = 4               # Once a record expires, how much longer (in
+                                 # seconds) may the current stale record be 
+                                 # used while the refresh query is executed in
+                                 # the background.
```

Comment:
yes. So while the record is valid (within ttl) it is just being used. Once it expires, (stale) it will trigger a background re-query. Once that requery completes, the record is valid again. So this setting is the maximum time that a record will be used while being stale. When it is refreshed it is no longer stale, and when it doesn't refresh within this `stale ttl` time, then requests will no longer use the stale record (it will effectively be deleted by then from the cache), but will synchronously wait for completion of the dns query (which is similar to the very first query when there is no stale record yet, and requests will have to wait for the query to finish as well)